src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -156,6 +179,8 @@ private function formatEditableCombinationsForListing(
                 );
             }
 
+            $imageData = Image::getBestImageAttribute($this->contextShopId, $langId, $productId, $combinationId);
```

Comment:
Not fan of using the legacy class here, could you create some endpoints in `ImageRepository` instead? We're gonna need some anyway to get the image list for the form later

Besides, this will allow to clean the `contextShopId` from this handler, we don't handle multishop much for now so I don't think you need to handle it in the repository for now It will be integrated in 1.7.9 (with lots of modifications and breaks I guess 😅 )